python/run_test.sh

```diff
@@ -3,6 +3,12 @@ unset SPARK_HOME
 
 python ../ci/lint_python.py --format --type-check || exit 1
 
+total_num_gpus=```nvidia-smi --query-gpu=index --format=csv,noheader | wc -l```
```

Comment:
I would use single back-quote, e.g.
```
total_num_gpus=`nvidia-smi --query-gpu=index --format=csv,noheader | wc -l`
```
or even better, use the parenthesis variant:
```
total_num_gpus=$(nvidia-smi --query-gpu=index --format=csv,noheader | wc -l)
```

Reply:
It seems this works:  `python -c 'import cupy; print(cupy.cuda.runtime.getDeviceCount())` and it respects CUDA_VISIBLE_DEVICES.

Reply:
Revised to "python -c ...." and used parenthesis.